ClinGen/clincoded
# Connects to #453. Connects to #479. Connects to #215. Connects to #488. Connects to #457

_[Edit: the links that follow aren’t real links — just to show where there are links]_

### Testing Notes

1. Make a new GDM and add a PMID.
2. Click the home button and verify two history entries in reverse chronological order (opposite order from this comment): PMID added and GDM created.
    - GDM created: [**Gene-Disease**-*mode*](#) *(link to Record Curation page for that GDM)* created; [date]
    - PMID added: [PMID:[id]](#) *(link to Record Curation page for the GDM with this PMID selected)* added to **Gene-Disease**-*Mode*; [date]
3. Create a group. In the user history, verify a group entry:
    - Group [group name](#) *(link to group view page)* added to **Gene-Disease**-*Mode* for [PMID:[id]](#) *(link to Record Curation page for the GDM with this PMID selected)*; [date]
4. Add a family to the group, with a variant and starter individual. Verify variant, family, and individual addition history items.
    - Variant [**variant ID**] added; [date]
    - Family [[family name](#)] (link to family view page) added to group [[group name](#)] (link to group view page) for [PMID:[id]](#) *(link to Record Curation page for the GDM with this PMID selected)*; [date]
    - Individual [[individual name](#)] (link to individual view page) added to family [[family name](#)] (link to family view page) for [PMID:[id]](#) *(link to Record Curation page for the GDM with this PMID selected)*; [date]
5. From the Record Curation page, curate the variant, but don’t assess it. Verify a variant pathogenicity entry.
    - Variant [[variant ID](#)] (link to variant pathogenicity edit page) pathogenicity added; [date]
6. Add experimental data, type Expression, “Altered expression in Patients,” “Is expression altered in patients who have the disease?” checked. Add a different variant from the earlier one, and assess the experimental data, all on the initiating page. Verify variant, experimental data addition, experimental data assessment user history entries
    - Variant [**variant ID**] added; [date]
    - Experimental data [[experimental data name](#)] (link to experimental data view page) ([experimental data type]) added to **Gene-Disease**-*Mode* for [PMID:[id]](#) *(link to Record Curation page for the GDM with this PMID selected)*; [date]
    - Experimental data [[experimental data name](#)] assessed as [**assessment value**]; [date]
7. Edit the first variant you made (the one with the starter individual) and assess it for Supports. Verify a variant modified and assessment history entry.
    - Variant [[variant id](#)] pathogenicity modified; [date]
    - Variant [[variant id](#)] pathogenicity assessed as **Supports**; [date]
8. From Record Curation, click the Generate Summary button. On the “Curation Summary & Provisional Classification” page, choose a classification besides “Limited” and save it. Verify a provisional classification entry:
    - [Provisional classification](#) *(link to classification edit page)** [CLASSIFICATION] added to **Gene-Disease**-*Mode*; [date]
9. From Record Curation, edit the group you created and delete it. Verify that entry in user history
    - Group [group name] deleted along with any associated families and individuals; [date]

### Documentation

Check the docs directory for curator_history.md that describes how modules can use the curator history module. Verify it makes sense and has good grammar.
